src/app/cakes/cakes.class.ts

```diff
@@ -0,0 +1,73 @@
+import { Status } from "./cakes.constants";
+
+export class Cake {
+  id: number;
+  name: string;
+  description: string;
+  ingredients: string[];
+  price: number;
+  stock: number;
+  status: number;
+
+  constructor(
+    id: number,
+    name: string,
+    description: string,
+    ingredients: string[],
+    price: number,
+    stock: number
+  ) {
+    this.id = id;
+    this.name = name;
+    this.description = description;
+    this.ingredients = ingredients;
+    this.price = price;
+    this.stock = stock;
+    this.status = Status.Available;
+  }
+
+  setStatus() {
```

Comment:
`computeStatus` could be a better name since the `set` verb expects a value in programming languages.